### ISSUE TITLE
CSP spec not user-friendly

### ISSUE THREAD

**Issue by galund** [w3c/webappsec-csp#673]
The content security policy recommendations are very unfriendly from the point of view of a developer/content publisher that needs to understand how to choose an appropriate policy. Perhaps the level 3 work is an opportunity to remedy this.

The current specification are excellent from a browser implementation perspective. But this has come at the expense of the context that would have been provided in an equivalent IETF RFC etc. Furthermore the examples do not speak enough to the kinds of security context in which they are appropriate, for example in choosing how strict the policy should be, or between `nonce` and hash-based security.

**Comment by Synchro** [comment]
I'm running into exactly this issue when trying to read the docs to parse CSP reports. The biggest problem is a lack of examples – the most verbose and useful example in the whole document is [this one](https://w3c.github.io/webappsec-csp/#example-f176bd19), but I can't help but notice that it includes a field named `document_url`, which is unlike the format used in the rest of the document! There is a `documentURL` property described in the JS structure, but nowhere do I see a mention of the mapping between these two representations of the same name. Ironically, [the description for constructing a deprecated CSP2 report](https://w3c.github.io/webappsec-csp/#deprecated-serialize-violation) provides useful, exact, explicit guidance on the precise field names, but this seems to be entirely missing from descriptions of CSP3 reports.

In practice, I see that Chrome does use the `documentURL` field name format, so perhaps that good example is outdated. Otherwise, is there some documented mapping between the formats of these field names?

In short, more examples, and more explicit guidance for implementation please.
